### PR TITLE
Feature/errors

### PR DESCRIPTION
**Pull request by nhweston**
This PR introduces Cooma runtime exceptions. These exceptions and utility methods for throwing them are defined in `./commons/…/scala/…/CoomaException.scala`. The error value has been removed, as have all calls to `sys.error` in interpreter code.

Interpreter entry points are wrapped in a `Try`. If a `CoomaException` is thrown, the error message is output. `CoomaException` is the only exception an interpreter is allowed to throw. If it throw any other exception, this is considered to be a bug in Cooma and an error message defined in `CoomaException.getUnhandledMessage` is output, along with an internal stack trace.

### CODE REVIEW

**Comment by nhweston** [comment]
Looks like it can't get the ScalaCheck library?

**Comment by inkytonik** [comment]
It seems to be this dependence which is failing:

```
"wolfendale" %% "scalacheck-gen-regexp" % "0.1.2"
```

My guess is that it used to be published on `bintray`. They recently stopped hosting some Scala stuff so that might explain why it's not working now. It works on our machines because we've previously downloaded the jar etc.

According to [Maven central](https://search.maven.org/artifact/io.github.wolfendale/scalacheck-gen-regexp_2.13/0.1.3/jar) version 0.1.3 of that library is published there, so can you try bumping the version and see if that fixes it on GH?

**Comment by nhweston** [comment]
This works:
```
"io.github.wolfendale" %% "scalacheck-gen-regexp" % "0.1.3"
```

File: src/test/scala/org/bitbucket/inkytonik/cooma/test/ExecutionTests.scala [hunk 1]
```diff
@@ -83,10 +83,10 @@ trait ExecutionTests extends REPLDriver with TestCompilerWithConfig[ASTNode, Pro
 
     def runBadPrimTest(name : String, args : String, error : String)(implicit bc : BackendConfig) : Unit = {
         val code = s"$name($args)"
-        val result1 = runString(name, code, Seq())
-        result1 shouldBe s"$error\n"
-        val result2 = runString(name, code, Seq("-r"))
-        result2 shouldBe s"$error\n"
+        //        val result1 = runString(name, code, Seq())
```
Comment (inkytonik) [inline src/test/scala/org/bitbucket/inkytonik/cooma/test/ExecutionTests.scala:86]: There's some commented out test code here...

**Review by inkytonik — COMMENTED** [review]
Looks really good except for the commented out test code in `ExecutionTests`.

**Comment by nhweston** [comment]
Getting some failing tests in `FileIoTests` after uncommenting.

**Comment by nhweston** [comment]
Actually, it seems these tests are failing regardless? Very strange.

**Comment by nhweston** [comment]
@inkytonik Ready to merge.

**Pull request merged**